### PR TITLE
Optionally limit flow on all upgraded proxy connections

### PR DESCRIPTION
**Pull request by brendandburns**
@lavalamp @dchen1107 @ncdc @smarterclayton

### CODE REVIEW

File: cmd/kube-apiserver/app/server.go [hunk 1]
```diff
@@ -205,6 +206,7 @@ func (s *APIServer) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&s.LongRunningRequestRE, "long-running-request-regexp", defaultLongRunningRequestRE, "A regular expression matching long running requests which should be excluded from maximum inflight request handling.")
 	fs.StringVar(&s.SSHUser, "ssh-user", "", "If non-empty, use secure SSH proxy to the nodes, using this user name")
 	fs.StringVar(&s.SSHKeyfile, "ssh-keyfile", "", "If non-empty, use secure SSH proxy to the nodes, using this user keyfile")
+	fs.Int64Var(&s.MaxConnectionBytesPerSec, "max-connection-bytes-per-sec", 0, "If non-zero, throttle each user connection to this number of bytes/sec.  Currently only applies to long-running requests")
```
Comment (thockin) [inline cmd/kube-apiserver/app/server.go]: General question: Shouldn't we be bounds-checking lags like this so they can't be negative?

Comment (brendandburns) [inline cmd/kube-apiserver/app/server.go]: The checks in proxy.go are `maxBytes > 0` so we catch it there.  I suppose we could pass an explicit flag validation warning somewhere, but seems like overkill...

**Comment by brendandburns** [comment]
@thockin @lavalamp 
friendly ping on this one.

File: pkg/registry/generic/rest/proxy.go [hunk 2]
```diff
@@ -54,6 +57,7 @@ func NewUpgradeAwareProxyHandler(location *url.URL, transport http.RoundTripper,
 		Transport:       transport,
 		UpgradeRequired: upgradeRequired,
 		FlushInterval:   defaultFlushInterval,
+		MaxBytesPerSec:  capabilities.Get().PerConnectionBandwidthLimitBytesPerSec,
```
Comment (smarterclayton) [inline pkg/registry/generic/rest/proxy.go]: This should really be an argument to the proxy handler from the rest registry code - I don't like the generic stuff being aware of a global static capabilities object (i'm less concerned with Kube specific code being aware of that).

This is more of a utility class than a Kube specific class - anyone using this in non Kubey ways may want different behavior.  If you have a high bandwidth proxy connection that shouldn't be limited this approach forces you to bypass the default).

Comment (brendandburns) [inline pkg/registry/generic/rest/proxy.go]: ok, moved it into the pod etcd registry.

**Comment by brendandburns** [comment]
Comment addressed, please take another look.

**Comment by smarterclayton** [comment]
Happier, LGTM

**Comment by brendandburns** [comment]
@k8s-bot test this please

**Comment by brendandburns** [comment]
@k8s-bot test this please

**Pull request merged**